src/wazuh_db/wdb_parser.c

```diff
@@ -6149,14 +6149,20 @@ int wdb_parse_agents_vuln_cves_update_status(wdb_t* wdb, char* input, char* outp
     else {
         cJSON* old_status = cJSON_GetObjectItem(data, "old_status");
         cJSON* new_status = cJSON_GetObjectItem(data, "new_status");
-        // Required fields
-        if (!cJSON_IsString(old_status) || !cJSON_IsString(new_status)) {
-            mdebug1("Invalid vuln_cves JSON data when updating status value. Not compliant with constraints defined in the database.");
-            snprintf(output, OS_MAXSTR + 1, "err Invalid JSON data, missing required fields");
-        }
+        cJSON* type = cJSON_GetObjectItem(data, "type");
 
-        else {
-            ret = wdb_agents_update_status_vuln_cves(wdb, old_status->valuestring, new_status->valuestring);
+        if (cJSON_IsString(type) && cJSON_IsString(new_status) && !cJSON_IsString(old_status)) {
+            ret = wdb_agents_update_status_vuln_cves(wdb, NULL, new_status->valuestring, type->valuestring);
```

Comment:
We can simplify this by obtaining NULL or valuestring for each parameter before, and making only one call with only one error handling:
```
const char new_status_str = cJSON_IsString(new_status) ? new_status->valuestring : NULL;
const char old_status_str = cJSON_IsString(old_status) ? old_status->valuestring : NULL;
const char type_str = cJSON_IsString(type) ? type->valuestring : NULL;
wdb_agents_update_status_vuln_cves(wdb, old_status_str, new_status_str, type_str);
```

or even simpler: Probably valuestring is **NULL** when **!cJSON_IsString()** and we can use the value directly

Reply:
Thank you. The method was updated